tests/backup/backup_test.go

```diff
@@ -334,6 +332,511 @@ var _ = Describe("{BackupCreateKillStoreRestore}", func() {
+									BackupRestoreCompletionTimeoutMin*time.Minute,
+									RetrySeconds*time.Second); err != nil {
+									errChan <- err
+									return
+								}
+
+								errChan <- err
+							})
+						}(provider, namespace)
+
+						go func(provider, namespace string) {
+							clusterName := fmt.Sprintf("%s-%s", ClusterName, provider)
+							restoreName := fmt.Sprintf("%s-%s-%s", RestoreNamePrefix, provider, namespace)
+							orgID := fmt.Sprintf("%s-%s-%s", strings.ToLower(taskNamePrefix),
+								provider, Inst().InstanceID)
+							Step(fmt.Sprintf("Delete restore full name %s:%s:%s",
+								clusterName, namespace, restoreName), func() {
+								backupDriver := Inst().Backup
+								deleteRestoreReq := &api.RestoreDeleteRequest{
+									OrgId: orgID,
+									Name:  restoreName,
+								}
+								_, err := backupDriver.DeleteRestore(deleteRestoreReq)
+
+								ctx, _ := context.WithTimeout(context.Background(),
+									BackupRestoreCompletionTimeoutMin*time.Minute)
+
+								logrus.Infof("Wait for restore %s is deleted", restoreName)
+								if err := backupDriver.WaitForRestoreDeletion(ctx, restoreName, orgID,
+									BackupRestoreCompletionTimeoutMin*time.Minute,
+									RetrySeconds*time.Second); err != nil {
+									errChan <- err
+									return
+								}
+
+								errChan <- err
+							})
+						}(provider, namespace)
+					}
+
+					for i := 0; i < len(providerNamespaces)*2; i++ {
+						select {
+						case <-ctx.Done():
+							Expect(err).NotTo(HaveOccurred(),
+								fmt.Sprintf("Failed to complete backup for provider %s cluster. Error: [%v]", provider, ctx.Err()))
+						case err := <-errChan:
+							Expect(err).NotTo(HaveOccurred(),
+								fmt.Sprintf("Failed to complete backup for provider %s cluster. Error: [%v]", provider, err))
+						}
+					}
+				}
+			})
+		}
+
+		Step("teardown backup objects for test", func() {
+			for _, provider := range providers {
+				providerClusterConfigPath, err := getProviderClusterConfigPath(provider, kubeconfigList)
+				Expect(err).NotTo(HaveOccurred(),
+					fmt.Sprintf("Failed to get kubeconfig path for provider %s cluster. Error: [%v]", provider, err))
+				SetClusterContext(providerClusterConfigPath)
+
+				logrus.Infof("Run Setup backup with object store provider: %s", provider)
+				orgID := fmt.Sprintf("%s-%s-%s", strings.ToLower(taskNamePrefix), provider, Inst().InstanceID)
+				bucketName := fmt.Sprintf("%s-%s-%s", BucketNamePrefix, provider, Inst().InstanceID)
+				credName := fmt.Sprintf("%s-%s", CredName, provider)
+				backupLocation := fmt.Sprintf("%s-%s", BLocationName, provider)
+				clusterName := fmt.Sprintf("%s-%s", ClusterName, provider)
+
+				DeleteCluster(clusterName, orgID)
+				DeleteBackupLocation(backupLocation, orgID)
+				DeleteCloudCredential(credName, orgID)
+				DeleteBucket(provider, bucketName)
+			}
+		})
+	})
+})
+
+func killStork() {
+	ctx := &scheduler.Context{
```

Comment:
Can we use this function in existing BackupCreateKillStoreRestore test as well?

Reply:
this will be part of refactoring PTX-2580